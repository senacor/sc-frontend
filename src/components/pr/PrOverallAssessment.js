import React from 'react';
import { withStyles } from '@material-ui/core/styles/index';
import ListItem from '@material-ui/core/ListItem';
import List from '@material-ui/core/List';
import ListItemText from '@material-ui/core/ListItemText';
import TextField from '@material-ui/core/TextField';
import { connect } from 'react-redux';
import MenuItem from '@material-ui/core/MenuItem';
import Icon from '@material-ui/core/Icon';
import Select from '@material-ui/core/Select';
import FormControl from '@material-ui/core/FormControl';
import * as actions from '../../actions';
import { debounce } from '../../helper/debounce';

const styles = theme => ({
  nestedText: {
    paddingLeft: '20px',
    paddingRight: '27px'
  },
  bootstrapInput: {
    borderRadius: 4,
    backgroundColor: theme.palette.common.white,
    border: '1px solid #ced4da',
    fontSize: 16,
    padding: '10px 12px',
    '&:focus': {
      borderColor: '#80bdff',
      boxShadow: '0 0 0 0.2rem rgba(0,123,255,.25)'
    }
  },
  containerListItem: {
    display: 'flex'
  },
  description: {
    paddingLeft: '0',
    paddingRight: '30px'
  },
  nestedListItem: {
    paddingLeft: '20px',
    width: '80%'
  },
  nestedListItemFTF: {
    maxWidth: '80%'
  },
  icon: {
    color: theme.palette.primary['400']
  },
  iconWhite: {
    color: theme.palette.common.white
  },
  nestedNumber: {
    minWidth: '44%',
    justifyContent: 'flex-end'
  },
  nestedNumberTargetRole: {
    width: '95%'
  },
  formControl: {
    paddingRight: '0px',
    minWidth: '95%'
  }
});

class PrOverallAssessment extends React.Component {
  constructor(props) {
    super(props);
    const categoryFulfillment = 'FULFILLMENT_OF_REQUIREMENT';
    const categoryTargetRole = 'TARGET_ROLE';

    this.state = {
      ratingFulfillment: this.props.prById.prRatingSet.find(
        prRating => prRating.prRatingDescription === categoryFulfillment
      ).rating,
      ratingTargetRole: this.props.prById.prRatingSet.find(
        prRating => prRating.prRatingDescription === categoryTargetRole
      ).rating,
      comment: this.props.prById.prRatingSet.find(
        prRating => prRating.prRatingDescription === categoryFulfillment
      ).comment,
      prById: this.props.prById,
      categoryFulfillment: categoryFulfillment,
      categoryTargetRole: categoryTargetRole,
      is_expanded: false
    };
  }

  handleChangeRating = prById => event => {
    this.setState({ [event.target.name]: event.target.value });
    let category = (event.target.name = 'ratingFulfillment'
      ? this.state.categoryFulfillment
      : this.state.categoryTargetRole);
    let prRating = prById.prRatingSet.find(
      prRating => prRating.prRatingDescription === category
    );
    this.props.addRating(
      prById,
      category,
      this.state.comment,
      event.target.value,
      prRating.id
    );
  };

  handleChangeComment = prById => event => {
    this.setState({ [event.target.name]: event.target.value });
    let prRating = prById.prRatingSet.find(
      prRating =>
        prRating.prRatingDescription === this.state.categoryFulfillment
    );

    this.sendComment(
      prById,
      this.state.categoryFulfillment,
      event.target.value,
      this.state.ratingFulfillment,
      prRating.id
    );
  };

  sendComment = debounce(this.props.addRating, 500);

  translateRatingDescription = ratingDescription => {
    switch (ratingDescription) {
      case 'FULFILLMENT_OF_REQUIREMENT':
        return 'Gesamteinschätzung; Erfüllung der Anforderungen für aktuelle Stufe';
      case 'TARGET_ROLE':
        return 'Zielrolle (ab Senior Level ausfüllen)';
      default:
        return 'default';
    }
  };

  categoryText = {
    text:
      'In welchem Umfang erfüllt die Mitarbeiterin/der Mitarbeiter die Anforderungen an seine aktuelle Laufbahnstufe vor dem Hintergrund der aktuellen Einstufung? Welche Stärken gilt es auszubauen, welche Lücken sollten geschlossen werden?'
  };

  render() {
    const { prById, classes } = this.props;

    return (
      <div>
        <div className={classes.containerListItem}>
          <ListItem>
            <ListItemText
              secondary={this.translateRatingDescription(
                this.state.categoryFulfillment
              )}
            />

            {this.state.comment ? (
              <Icon className={classes.icon}>comment</Icon>
            ) : (
              <Icon className={classes.iconWhite}>comment</Icon>
            )}
          </ListItem>

          <ListItem className={classes.nestedNumber}>
            <FormControl className={classes.formControl}>
              <Select
                value={
                  this.state.ratingFulfillment
                    ? this.state.ratingFulfillment
                    : 3
                }
                onChange={this.handleChangeRating(prById)}
                displayEmpty
                name="ratingFulfillment"
              >
                <MenuItem value={1}>nicht erfüllt</MenuItem>,
                <MenuItem value={2}>zT. n. erfüllt</MenuItem>,
                <MenuItem value={3}>erfüllt</MenuItem>,
                <MenuItem value={4}>zT. übererfüllt</MenuItem>,
                <MenuItem value={5}>übererfüllt</MenuItem>
              </Select>
            </FormControl>
          </ListItem>
        </div>
        <List component="div" disablePadding className={classes.nestedText}>
          <ListItem>
            <ListItemText
              secondary={this.translateRatingDescription(
                this.state.categoryTargetRole
              )}
            />
            <FormControl className={classes.nestedNumberTargetRole}>
              <Select
                value={
                  this.state.ratingTargetRole ? this.state.ratingTargetRole : 1
                }
                onChange={this.handleChangeRating(prById)}
                displayEmpty
                name="ratingTargetRole"
              >
                <MenuItem value={1}>keine Auswahl</MenuItem>,
                <MenuItem value={2}>Platformgestalter</MenuItem>,
                <MenuItem value={3}>Business IT Solution Leader</MenuItem>,
                <MenuItem value={4}>Transformation Manager</MenuItem>,
                <MenuItem value={5}>IT-Liefersteuerer</MenuItem>,
                <MenuItem value={6}>Architect</MenuItem>,
                <MenuItem value={7}>Technical Expert</MenuItem>,
                <MenuItem value={8}>Lead Developer</MenuItem>
              </Select>
            </FormControl>
          </ListItem>
          <ListItem>
            <TextField
              id={this.state.categoryFulfillment}
              label="Freitextfeld (bitte ausfüllen)"
              multiline
              fullWidth
              rowsMax="4"
              value={this.state.comment ? this.state.comment : ''}
              onChange={this.handleChangeComment(prById)}
              InputProps={{
                disableUnderline: true,
                name: 'comment',
                classes: {
                  input: classes.bootstrapInput
                }
              }}
              InputLabelProps={{
                shrink: true
              }}
            />
          </ListItem>
          <ListItem />
          <ListItem>
            <ListItemText
              className={classes.description}
              primary="Beschreibung"
              secondary={this.categoryText.text}
            />
          </ListItem>
        </List>
      </div>
    );
  }
}

export const StyledComponentOA = withStyles(styles)(PrOverallAssessment);
export default connect(
  state => ({
    prRating: state.prRatings.prRating
  }),
  {
    addRating: actions.addRating
  }
)(StyledComponentOA);
