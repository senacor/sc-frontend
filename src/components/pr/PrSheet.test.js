import React from 'react';
import { StyledComponent } from './PrSheet';
import { default as PrOverallAssessment } from './PrOverallAssessment';
import { createShallow } from '@material-ui/core/test-utils';
import ROLES from '../../helper/roles';

describe('PrSheet Component', () => {
  let shallow = createShallow({ dive: true });

  const prById = {
    id: 1,
    employee: {
      id: 1,
      login: 'lschäfer',
      firstName: 'Lionel',
      lastName: 'Schäfer',
      title: 'DR',
      email: 'lionel.schäfer@senacor.com',
      entryDate: '2004-05-10',
      salaries: [],
      employment: {}
    },
    supervisor: 'ttran',
    occasion: 'ON_DEMAND',
    status: 'PREPARATION',
    deadline: '2015-05-11',
    statuses: [
      'FILLED_SHEET_EMPLOYEE',
      'FILLED_SHEET_REVIEWER',
      'ALL_DATES_ACCEPTED'
    ],
    _links: {
      self: {
        href: 'http://localhost:8010/api/v1/prs/1'
      }
    },
    prRatingSet: [
      {
        id: 29,
        prRatingDescription: 'FULFILLMENT_OF_REQUIREMENT',
        prRatingCategory: 'OVERALL_ASSESSMENT',
        rating: 3,
        comment: 'Everything is awesome!!!',
        _links: {
          self: {
            href: 'http://localhost:8010/api/v1/prs/3/ratings/29'
          }
        }
      },
      {
        id: 22,
        prRatingDescription: 'WORK_RESULTS',
        prRatingCategory: 'PERFORMANCE_IN_PROJECT',
        rating: 4,
        comment: null,
        _links: {
          self: {
            href: 'http://localhost:8010/api/v1/prs/3/ratings/22'
          }
        }
      },
      {
        id: 24,
        prRatingDescription: 'CUSTOMER_INTERACTION',
        prRatingCategory: 'IMPACT_ON_CUSTOMER',
        rating: 5,
        comment: 'Employee uses Attract. It is super effective!',
        _links: {
          self: {
            href: 'http://localhost:8010/api/v1/prs/3/ratings/24'
          }
        }
      },
      {
        id: 30,
        prRatingDescription: 'TARGET_ROLE',
        prRatingCategory: 'OVERALL_ASSESSMENT',
        rating: 2,
        comment: null,
        _links: {
          self: {
            href: 'http://localhost:8010/api/v1/prs/3/ratings/30'
          }
        }
      },
      {
        id: 26,
        prRatingDescription: 'LEADERSHIP',
        prRatingCategory: 'IMPACT_ON_TEAM',
        rating: 3,
        comment: null,
        _links: {
          self: {
            href: 'http://localhost:8010/api/v1/prs/3/ratings/26'
          }
        }
      },
      {
        id: 27,
        prRatingDescription: 'TEAMWORK',
        prRatingCategory: 'IMPACT_ON_TEAM',
        rating: 3,
        comment: null,
        _links: {
          self: {
            href: 'http://localhost:8010/api/v1/prs/3/ratings/27'
          }
        }
      },
      {
        id: 21,
        prRatingDescription: 'PROBLEM_ANALYSIS',
        prRatingCategory: 'PERFORMANCE_IN_PROJECT',
        rating: 4,
        comment: null,
        _links: {
          self: {
            href: 'http://localhost:8010/api/v1/prs/3/ratings/21'
          }
        }
      },
      {
        id: 23,
        prRatingDescription: 'WORKING_MANNER',
        prRatingCategory: 'PERFORMANCE_IN_PROJECT',
        rating: 2,
        comment: null,
        _links: {
          self: {
            href: 'http://localhost:8010/api/v1/prs/3/ratings/23'
          }
        }
      },
      {
        id: 25,
        prRatingDescription: 'CUSTOMER_RETENTION',
        prRatingCategory: 'IMPACT_ON_CUSTOMER',
        rating: 5,
        comment: null,
        _links: {
          self: {
            href: 'http://localhost:8010/api/v1/prs/3/ratings/25'
          }
        }
      },
      {
        id: 28,
        prRatingDescription: 'CONTRIBUTION_TO_COMPANY_DEVELOPMENT',
        prRatingCategory: 'IMPACT_ON_COMPANY',
        rating: 3,
        comment: null,
        _links: {
          self: {
            href: 'http://localhost:8010/api/v1/prs/3/ratings/28'
          }
        }
      }
    ],
    prReflectionSet: [
      {
        id: 5,
        prReflectionField: 'INFLUENCE_OF_LEADER_AND_ENVIRONMENT',
        text:
          'We went on a hike together and my leader showed me the mountains',
        _links: {
          self: {
            href: 'http://localhost:8010/api/v1/prs/3/reflections/5'
          }
        }
      },
      {
        id: 6,
        prReflectionField: 'ROLE_AND_PROJECT_ENVIRONMENT',
        text: 'I am a hiker now',
        _links: {
          self: {
            href: 'http://localhost:8010/api/v1/prs/3/reflections/6'
          }
        }
      }
    ],
    prVisibilityEntry: {
      visibilityToEmployee: 'INVISIBLE',
      visibilityToReviewer: 'VISIBLE'
    },
    prFinalizationStatus: {
      finalizationStatusOfEmployee: 'NOT_FINALIZED',
      finalizationStatusOfReviewer: 'NOT_FINALIZED'
    }
  };

  it('should display the PrSheet', () => {
    const fetchVisibilityMock = jest.fn();
    const component = shallow(
      <StyledComponent
        prById={prById}
        fetchPrVisibilityById={fetchVisibilityMock}
        userroles={[ROLES.PR_MITARBEITER]}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it("should not show the supervisor's comment or ratings", () => {
    const fetchVisibilityMock = jest.fn();
    const component = shallow(
      <StyledComponent
        prById={prById}
        fetchPrVisibilityById={fetchVisibilityMock}
        userroles={[ROLES.PR_MITARBEITER]}
      />
    );
    expect(
      component.contains(
        <PrOverallAssessment
          prById={prById}
          prVisible={false}
          prFinalized={false}
        />
      )
    ).toBe(true);
    expect(
      component.find('Connect(WithStyles(PrComment))[prVisible=false]')
    ).toHaveLength(8);
  });

  it("should show the employees's contributions", () => {
    const fetchVisibilityMock = jest.fn();
    const component = shallow(
      <StyledComponent
        prById={prById}
        fetchPrVisibilityById={fetchVisibilityMock}
        userroles={[ROLES.PR_MITARBEITER]}
      />
    );

    expect(
      component.find('Connect(WithStyles(PrSheetEmployee))[prVisible=true]')
    ).toHaveLength(2);
  });

  it("should display the supervisor's comment or ratings as readonly", () => {
    let statuses = {
      statuses: [
        'FILLED_SHEET_EMPLOYEE',
        'FILLED_SHEET_REVIEWER',
        'ALL_DATES_ACCEPTED',
        'MODIFICATIONS_ACCEPTED_REVIEWER'
      ]
    };

    let finalization = {
      prFinalizationStatus: {
        finalizationStatusOfEmployee: 'NOT_FINALIZED',
        finalizationStatusOfReviewer: 'FINALIZED'
      }
    };
    let prByIdWhichReviewerFinalized = Object.assign(
      prById,
      statuses,
      finalization
    );
    const fetchVisibilityMock = jest.fn();
    const component = shallow(
      <StyledComponent
        prById={prByIdWhichReviewerFinalized}
        fetchPrVisibilityById={fetchVisibilityMock}
        userroles={[ROLES.PR_CST_LEITER]}
      />
    );

    expect(
      component.contains(
        <PrOverallAssessment
          prById={prById}
          prVisible={true}
          prFinalized={true}
        />
      )
    ).toBe(true);

    expect(
      component.find('Connect(WithStyles(PrComment))[prFinalized=true]')
    ).toHaveLength(8);
  });
});
