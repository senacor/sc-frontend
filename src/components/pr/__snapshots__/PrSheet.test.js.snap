// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrSheet Component displays the PrSheet 1`] = `
<div>
  <div
    className="PrSheet-containerVertical-1"
  >
    <WithStyles(List)>
      <WithStyles(ListItem)>
        <WithStyles(ListItemText)
          primary="Leistungen im Projekt"
        />
      </WithStyles(ListItem)>
      <WithStyles(List)
        disablePadding={true}
      >
        <WithStyles(ListItem)
          button={true}
          className="PrSheet-nested-2"
          onClick={[Function]}
        >
          <WithStyles(ListItemText)
            secondary="Problemanalyse und -lösung"
          />
          <WithStyles(Icon)>
            expand_more
          </WithStyles(Icon)>
          <TextField
            className=""
            onChange={[Function]}
            required={false}
            select={true}
            value="1"
          >
            <WithStyles(MenuItem)
              key="1"
              value="1"
            >
              1
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              key="2"
              value="2"
            >
              2
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              key="3"
              value="3"
            >
              3
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              key="4"
              value="4"
            >
              4
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              key="5"
              value="5"
            >
              5
            </WithStyles(MenuItem)>
          </TextField>
        </WithStyles(ListItem)>
        <WithStyles(Collapse)
          in={false}
          timeout="auto"
          unmountOnExit={true}
        >
          <WithStyles(List)
            className="PrSheet-nestedText-3"
            component="div"
            disablePadding={true}
          >
            <WithStyles(ListItem)>
              <TextField
                InputLabelProps={
                  Object {
                    "className": undefined,
                    "shrink": true,
                  }
                }
                InputProps={
                  Object {
                    "classes": Object {
                      "input": "PrSheet-bootstrapInput-6",
                      "root": "PrSheet-bootstrapRoot-5",
                    },
                    "disableUnderline": true,
                  }
                }
                fullWidth={true}
                id="multiline-flexible"
                label="Kommentar"
                multiline={true}
                onChange={[Function]}
                required={false}
                rowsMax="4"
                select={false}
              />
            </WithStyles(ListItem)>
            <WithStyles(ListItem) />
            <WithStyles(ListItem)
              button={true}
            >
              <WithStyles(ListItemText)
                disabled={true}
                secondary="Beschreibung: Problemanalyse !"
              />
            </WithStyles(ListItem)>
          </WithStyles(List)>
          <WithStyles(Divider) />
        </WithStyles(Collapse)>
        <WithStyles(ListItem)
          button={true}
          className="PrSheet-nested-2"
          onClick={[Function]}
        >
          <WithStyles(ListItemText)
            secondary="Arbeitsergebnisse"
          />
          <WithStyles(Icon)>
            expand_more
          </WithStyles(Icon)>
          <TextField
            className=""
            onChange={[Function]}
            required={false}
            select={true}
            value="1"
          >
            <WithStyles(MenuItem)
              key="1"
              value="1"
            >
              1
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              key="2"
              value="2"
            >
              2
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              key="3"
              value="3"
            >
              3
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              key="4"
              value="4"
            >
              4
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              key="5"
              value="5"
            >
              5
            </WithStyles(MenuItem)>
          </TextField>
        </WithStyles(ListItem)>
        <WithStyles(Collapse)
          in={false}
          timeout="auto"
          unmountOnExit={true}
        >
          <WithStyles(List)
            className="PrSheet-nestedText-3"
            component="div"
            disablePadding={true}
          >
            <WithStyles(ListItem)>
              <TextField
                InputLabelProps={
                  Object {
                    "className": undefined,
                    "shrink": true,
                  }
                }
                InputProps={
                  Object {
                    "classes": Object {
                      "input": "PrSheet-bootstrapInput-6",
                      "root": "PrSheet-bootstrapRoot-5",
                    },
                    "disableUnderline": true,
                  }
                }
                fullWidth={true}
                id="multiline-flexible"
                label="Kommentar"
                multiline={true}
                onChange={[Function]}
                required={false}
                rowsMax="4"
                select={false}
              />
            </WithStyles(ListItem)>
            <WithStyles(ListItem) />
            <WithStyles(ListItem)
              button={true}
            >
              <WithStyles(ListItemText)
                disabled={true}
                secondary="Beschreibung:  Arbeitsergebnisse !"
              />
            </WithStyles(ListItem)>
          </WithStyles(List)>
          <WithStyles(Divider) />
        </WithStyles(Collapse)>
        <WithStyles(ListItem)
          button={true}
          className="PrSheet-nested-2"
          onClick={[Function]}
        >
          <WithStyles(ListItemText)
            secondary="Arbeitsweise"
          />
          <WithStyles(Icon)>
            expand_more
          </WithStyles(Icon)>
          <TextField
            className=""
            onChange={[Function]}
            required={false}
            select={true}
            value="1"
          >
            <WithStyles(MenuItem)
              key="1"
              value="1"
            >
              1
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              key="2"
              value="2"
            >
              2
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              key="3"
              value="3"
            >
              3
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              key="4"
              value="4"
            >
              4
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              key="5"
              value="5"
            >
              5
            </WithStyles(MenuItem)>
          </TextField>
        </WithStyles(ListItem)>
        <WithStyles(Collapse)
          in={false}
          timeout="auto"
          unmountOnExit={true}
        >
          <WithStyles(List)
            className="PrSheet-nestedText-3"
            component="div"
            disablePadding={true}
          >
            <WithStyles(ListItem)>
              <TextField
                InputLabelProps={
                  Object {
                    "className": undefined,
                    "shrink": true,
                  }
                }
                InputProps={
                  Object {
                    "classes": Object {
                      "input": "PrSheet-bootstrapInput-6",
                      "root": "PrSheet-bootstrapRoot-5",
                    },
                    "disableUnderline": true,
                  }
                }
                fullWidth={true}
                id="multiline-flexible"
                label="Kommentar"
                multiline={true}
                onChange={[Function]}
                required={false}
                rowsMax="4"
                select={false}
              />
            </WithStyles(ListItem)>
            <WithStyles(ListItem) />
            <WithStyles(ListItem)
              button={true}
            >
              <WithStyles(ListItemText)
                disabled={true}
                secondary="Beschreibung: Arbeitsweise !"
              />
            </WithStyles(ListItem)>
          </WithStyles(List)>
          <WithStyles(Divider) />
        </WithStyles(Collapse)>
      </WithStyles(List)>
    </WithStyles(List)>
    <WithStyles(Divider) />
    <WithStyles(List)>
      <WithStyles(ListItem)>
        <WithStyles(ListItemText)
          primary="Wirkung beim Kunden"
        />
      </WithStyles(ListItem)>
      <WithStyles(List)
        disablePadding={true}
      >
        <WithStyles(ListItem)
          button={true}
          className="PrSheet-nested-2"
        >
          <WithStyles(ListItemText)
            secondary="Kundeninteraktion und -veränderung"
          />
        </WithStyles(ListItem)>
        <WithStyles(ListItem)
          button={true}
          className="PrSheet-nested-2"
        >
          <WithStyles(ListItemText)
            secondary="Kundenbindung und Mandatsgenerierung"
          />
        </WithStyles(ListItem)>
      </WithStyles(List)>
    </WithStyles(List)>
    <WithStyles(Divider) />
    <WithStyles(List)>
      <WithStyles(ListItem)>
        <WithStyles(ListItemText)
          primary="Wirkung im Team"
        />
      </WithStyles(ListItem)>
      <WithStyles(List)
        disablePadding={true}
      >
        <WithStyles(ListItem)
          button={true}
          className="PrSheet-nested-2"
        >
          <WithStyles(ListItemText)
            secondary="Effektives Arbeiten im Team und Teamführung"
          />
        </WithStyles(ListItem)>
        <WithStyles(ListItem)
          button={true}
          className="PrSheet-nested-2"
        >
          <WithStyles(ListItemText)
            secondary="Coaching, Leadership und Personalführung"
          />
        </WithStyles(ListItem)>
      </WithStyles(List)>
    </WithStyles(List)>
    <WithStyles(Divider) />
    <WithStyles(List)>
      <WithStyles(ListItem)>
        <WithStyles(ListItemText)
          primary="Wirkung im Unternehmen"
        />
      </WithStyles(ListItem)>
      <WithStyles(List)
        disablePadding={true}
      >
        <WithStyles(ListItem)
          button={true}
          className="PrSheet-nested-2"
        >
          <WithStyles(ListItemText)
            secondary="Beitrag zur Unternehmensentwicklung"
          />
        </WithStyles(ListItem)>
      </WithStyles(List)>
    </WithStyles(List)>
    <WithStyles(Divider) />
    <WithStyles(List)>
      <WithStyles(ListItem)>
        <WithStyles(ListItemText)
          primary="Gesamtschätzung und Entwicklungsbedarfe"
        />
      </WithStyles(ListItem)>
      <WithStyles(List)
        disablePadding={true}
      >
        <WithStyles(ListItem)
          button={true}
          className="PrSheet-nested-2"
        >
          <WithStyles(ListItemText)
            secondary="Freitextfeld"
          />
        </WithStyles(ListItem)>
      </WithStyles(List)>
    </WithStyles(List)>
    <WithStyles(Divider) />
  </div>
</div>
`;
